### PR TITLE
[FLOC-3869] Fix lint problems

### PR DESCRIPTION
**Pull request by jongiddy**
These lint problems are mostly:
1. imports appearing after other code
2. assigned lambdas
3. operators appearing after newline rather than before.

Most of the fixes are to shuffle code around a bit, and convert lambdas to functions.

In `flocker/node`, there was a circular import which worked previously because one of the imports was after the definition of an Exception object.  The exception was moved to its own file to break the circular import.

### CODE REVIEW

File: flocker/testtools/_flaky.py [hunk 1]
```diff
@@ -77,6 +77,13 @@ def _get_flaky_annotation(case):
     return getattr(method, _FLAKY_ATTRIBUTE, None)
 
 
+def _get_invariants(x):
```
Comment (jml) [inline flocker/testtools/_flaky.py]: Please rename to `_flaky_invariants`.

**Comment by jml** [comment]
Thanks! It's regrettable that you've lost your morning to this, and that @wallrj and @justnoise have lost a day of momentum. 

FWIW, I personally disagree with all of the "new" rules, but think (at least for these) that we should just lump it.

Please merge when the tests pass.

**Comment by jongiddy** [comment]
@jml Now that the lint test is green, should we add an issue to disable some checks? The restriction against lambda assignment, in particular, seems obnoxious, especially for class attributes.

**Comment by jml** [comment]
Yeah, I think so. lambda assignment for sure and maybe also the `__all__` / `import` thing.

**Pull request merged**